pkg/features/kube_features.go

```diff
@@ -765,6 +765,12 @@ const (
 	//
 	// Enables Leader Migration for kube-controller-manager and cloud-controller-manager
 	ControllerManagerLeaderMigration featuregate.Feature = "ControllerManagerLeaderMigration"
+
+	// owner: @adrianreber
+	// alpha: v1.23
+	//
+	// Enables Checkpoint/Restore in the kubelet
+	CheckpointRestore featuregate.Feature = "CheckpointRestore"
```

Comment:
Would this be confusing from the user's point of view? It's fine for long-term vision but for now, just checkpoint would be enabled with this PR but reading this seems like both Checkpoint and Restore functionalities are enabled.

Reply:
I also thought about it. Not sure. The long term goal is to also have restore support. Not sure if it makes sense to have two feature gates if we ever introduce restore support in the future. I opted for a combined feature gate name as it will be unlikely to have only checkpoint or only restore enabled. The corresponding documentation (does not yet exist) should help make it clear what the feature gate actually means. In the end it is just a string and I am open to changing it. The current name is just what made most sense to me.